via/views/route_by_content.py

```diff
@@ -12,15 +12,17 @@
     UnhandledException,
     UpstreamServiceError,
 )
-from via.views._query_params import strip_client_query_params
+from via.views._query_params import QueryParams
 
 
 @view.view_config(route_name="route_by_content")
 def route_by_content(request):
     """Routes the request according to the Content-Type header."""
-    url = strip_client_query_params(request.matchdict["url"], request.params)
```

Comment:
This `url` parameter is re-used later on and confused the heck out of me about when we do or don't strip via params

Reply:
Honestly just writing `request.matchdict["url"]` every time you want to use it, rather than assigning a variable name to it, would probably make the code easier to follow

Reply:
It actually is already referenced as `request.matchdict["url"]` in one place further down in the function, but referenced as `path_url` in others

Reply:
I'm a bit confused about what you're saying here, but to clarify:

 * Previously `url` was used twice, for different purposes obtained in different ways
 * We now get the `url` once, and use the same method for constructing the final URL with slight differences

Reply:
Ah I get it. There's a _third_ place I hadn't spotted. I've fixed this now.

In the first iteration of this there were about 4 different URLs floating about of which 2 had the same name etc.